src/Charcoal/Email/Services/Parser.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Charcoal\Email\Services;
+
+use InvalidArgumentException;
+
+/**
+ * Parser service
+ */
+class Parser
+{
+    /**
+     * @param mixed $email An email value (either a string or an array).
+     * @throws InvalidArgumentException If the email is invalid.
+     * @return string
+     */
+    public function parse($email): string
+    {
+        if (is_array($email)) {
+            return $this->emailFromArray($email);
+        } elseif (is_string($email)) {
+            $arr = $this->emailToArray($email);
+            return $this->emailFromArray($arr);
+        } else {
+            throw new InvalidArgumentException(
+                'Can not parse email: must be an array or a string'
+            );
+        }
+    }
+
+    /**
+     * Convert an email address (RFC822) into a proper array notation.
+     *
+     * @param  string|array $var An email array (containing an "email" key and optionally a "name" key).
+     * @throws InvalidArgumentException If the email is invalid.
+     * @return array
+     */
+    public function emailToArray(string $var) : array
+    {
+        $regexp = '/(.+[\s]+)?(<)?(([\w\-\._\+]+)@((?:[\w\-_]+\.)+)([a-zA-Z]*))?(>)?/u';
```

Comment:
Could you change this variable into a public class constant.

Would allow for simpler extension of the class. One could simply change the value of the constant.

Reply:
Can't really think of a use case but sure.